src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala

```diff
@@ -55,11 +57,20 @@ case class SmartDataLakeBuilderConfig(feedSel: String = null,
                                       partitionValues: Option[Seq[PartitionValues]] = None,
                                       multiPartitionValues: Option[Seq[PartitionValues]] = None,
                                       parallelism: Int = 1,
+                                      statePath: Option[String] = None,
+                                      runId: Option[Int] = None,
                                       overrideJars: Option[Seq[String]] = None
                                 ) {
   def validate(): Unit = {
-    assert(!master.contains("yarn") || deployMode.nonEmpty, "spark deploy-mode must be set if spark master=yarn")
-    assert(partitionValues.isEmpty || multiPartitionValues.isEmpty, "partitionValues and multiPartitionValues cannot be defined at the same time")
+    assert(!applicationName.exists(_.contains("_")), s"Application name must not contain character '_' ($applicationName)")
```

Comment:
Why are application names with `_` not allowed?

Reply:
We use "_" as delimiter in the state filename to separate appName_runId_attemptId.